kubernetes/cache/kube_cache.go

```diff
@@ -537,6 +538,53 @@ func (c *kubeCache) GetDaemonSet(namespace, name string) (*apps_v1.DaemonSet, er
 	return retDS, nil
 }
 
+func (c *kubeCache) GetDaemonSetsWithSelector(namespace string, selectorLabels string) ([]apps_v1.DaemonSet, error) {
+	defer c.cacheLock.RUnlock()
+	c.cacheLock.RLock()
+
+	var daemonSets []apps_v1.DaemonSet
+	var err error
+
+	selector, err := labels.ConvertSelectorToLabelsMap(selectorLabels)
+	if err != nil {
+		return nil, err
+	}
+
+	if namespace == metav1.NamespaceAll {
+		if c.clusterScoped {
+			daemonSets, err = c.GetDaemonSets(namespace)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
I think I see why the filtering isn't working. Calling `c.GetDaemonSets` here won't apply the `selector` since `c.GetDaemonSets` just passes `labels.Everything()`. You probably need to just copy these lines from that function so that you can pass the selector into the `List` call:
```
	daemonSets, err := c.getCacheLister(namespace).daemonSetLister.DaemonSets(namespace).List(labels.Everything())
	if err != nil {
		return nil, err
	}
```
rather than trying to call `c.GetDaemonSets` and then add the type/deep copy at the bottom.